packages/react-button/src/components/ToggleButton/useToggle.ts

```diff
@@ -10,22 +10,29 @@ interface ToggleProps {
 interface ToggleState extends ToggleProps {}
 
 /**
- * The useToggleButton hook processes the Button component props and returns state.
+ * The useToggleButton hook processes the ToggleButton component props and returns state.
  * @param props - ToggleButton props to derive state from.
  */
 export const useToggle = <TProps, TState extends TProps = TProps>(
   props: TProps & ToggleProps,
 ): TState & ToggleState => {
```

Comment:
I might have asked this before. why is `TState` needed here?

Reply:
Because we're going to be returning something that is more than just the `defaultChecked`, `checked`, `onClick` props we're using here. The idea is that we can use this `useToggle` hook on other components besides the `ToggleButton` later on.